graylog2-web-interface/src/components/common/HasOwnership.jsx

```diff
@@ -0,0 +1,42 @@
+// @flow strict
+import * as React from 'react';
+import { useContext } from 'react';
+import PropTypes from 'prop-types';
+
+import CurrentUserContext from 'contexts/CurrentUserContext';
+import { createGRN } from 'logic/permissions/GRN';
+
+type Props = {
+  children: React.Node,
+  id: string,
+  type: string,
+};
+
+const HasOwnership = ({ children, id, type }: Props) => {
+  const currentUser = useContext(CurrentUserContext);
+  const entity = createGRN(id, type);
+  const ownership = `entity:own:${entity}`;
+  const adminPermission = '*';
+
+  if (currentUser) {
+    const { grn_permissions: grnPermissions = [], permissions } = currentUser;
+    const isAdmin = permissions.includes(adminPermission);
+
+    if (grnPermissions.includes(ownership) || isAdmin) {
+      return children;
+    }
+  }
+
+  return null;
+};
+
+HasOwnership.propTypes = {
+  /**  Children to render if user has entity of entity */
```

Comment:
Do you mean something like `user has ownership of entity`? And let's remove the space before children?